src/platform/platform.dom.js

```diff
@@ -225,50 +191,54 @@ function watchForAttachment(element, fn) {
 	return observer;
 }
 
-/**
- * Watch for detachment of `element` from its direct `parent`.
- * @param {Element} element - The element to watch
- * @param {function} fn - Callback function to call when detached.
- * @return {MutationObserver=}
- */
-function watchForDetachment(element, fn) {
-	const parent = _getParentNode(element);
-	if (!parent) {
+function createDetachObserver(chart, type, listener) {
+	const canvas = chart.canvas;
+	const container = canvas && _getParentNode(canvas);
+	if (!container) {
 		return;
 	}
 	const observer = new MutationObserver(entries => {
 		entries.forEach(entry => {
 			for (let i = 0; i < entry.removedNodes.length; i++) {
-				if (entry.removedNodes[i] === element) {
-					fn();
+				if (entry.removedNodes[i] === canvas) {
+					listener();
 					break;
 				}
 			}
 		});
 	});
-	observer.observe(parent, {childList: true});
+	observer.observe(container, {childList: true});
 	return observer;
 }
 
-function createAttachObserver(chart, type, listener) {
-	const canvas = chart.canvas;
-	const container = canvas && _getParentNode(canvas);
-	return watchForAttachment(container || canvas, listener);
-}
-
-function createDetachObserver(chart, type, listener) {
-	const canvas = chart.canvas;
-	if (canvas) {
-		return watchForDetachment(canvas, listener);
-	}
-}
-
 function createResizeObserver(chart, type, listener) {
 	const canvas = chart.canvas;
 	const container = canvas && _getParentNode(canvas);
-	if (container) {
-		return watchForResize(container, listener);
+	if (!container) {
+		return;
 	}
+	const resize = throttled((width, height) => {
+		const w = container.clientWidth;
+		listener(width, height);
+		if (w < container.clientWidth) {
+			// If the container size shrank during chart resize, let's assume
+			// scrollbar appeared. So we resize again with the scrollbar visible -
+			// effectively making chart smaller and the scrollbar hidden again.
+			// Because we are inside `throttled`, and currently `ticking`, scroll
+			// events are ignored during this whole 2 resize process.
+			// If we assumed wrong and something else happened, we are resizing
+			// twice in a frame (potential performance issue)
+			listener();
+		}
+	}, window);
+
+	// @ts-ignore until https://github.com/Microsoft/TypeScript/issues/28502 implemented
```

Comment:
Can you update this URL to https://github.com/microsoft/TypeScript/issues/37861? There's a new issue tracking it now